src/vessel/program.clj

```diff
@@ -50,6 +51,12 @@
              :desc "Directory where the application's files will be written to"
              :parse-fn io/file
              :validate cli/file-or-dir-must-exist]
+            ["-C" "--compiler-options OPTIONS"
+             :id :compiler-options
+             :desc "Options provided to the Clojure compiler, see clojure.core/*compiler-options*"
+             :default nil
+             :parse-fn edn/read-string
+             :validate [#(or (nil? %) (map? %)) "Compiler options must be a valid Clojure map"]]
```

Comment:
💅  maybe it's worth moving this sexpr to a def since you are duplicating it below.